File: identhub/src/main/java/de/solarisbank/identhub/session/data/verification/phone/VerificationPhoneDataSourceRepository.java
package de.solarisbank.identhub.session.data.verification.phone;

import de.solarisbank.identhub.domain.verification.phone.VerificationPhoneRepository;
import de.solarisbank.identhub.session.data.TransactionAuthenticationNumber;
import de.solarisbank.identhub.session.data.verification.phone.model.VerificationPhoneResponse;
import io.reactivex.Single;
import io.reactivex.schedulers.Schedulers;

public class VerificationPhoneDataSourceRepository implements VerificationPhoneRepository {

    private final VerificationPhoneNetworkDataSource verificationPhoneNetworkDataSource;

    public VerificationPhoneDataSourceRepository(VerificationPhoneNetworkDataSource verificationPhoneNetworkDataSource) {
        this.verificationPhoneNetworkDataSource = verificationPhoneNetworkDataSource;
    }

    @Override
    public Single<VerificationPhoneResponse> authorize() {
        return verificationPhoneNetworkDataSource.postAuthorize()
                .subscribeOn(Schedulers.io());
    }

    @Override
    public Single<VerificationPhoneResponse> confirmToken(TransactionAuthenticationNumber token) {
        return verificationPhoneNetworkDataSource.postConfirm(token)
                .subscribeOn(Schedulers.io());
    }
}

File: phone/src/main/java/de/solarisbank/identhub/phone/data/VerificationPhoneDataSourceRepository.kt
package de.solarisbank.identhub.phone.data

import de.solarisbank.identhub.phone.domain.VerificationPhoneRepository
import de.solarisbank.identhub.phone.model.TransactionAuthenticationNumber
import de.solarisbank.identhub.phone.model.VerificationPhoneResponse
import io.reactivex.Single
import io.reactivex.schedulers.Schedulers

class VerificationPhoneDataSourceRepository(
    private val verificationPhoneNetworkDataSource: VerificationPhoneNetworkDataSource
    ) : VerificationPhoneRepository {
    override fun authorize(): Single<VerificationPhoneResponse> {
        return verificationPhoneNetworkDataSource.postAuthorize()
            .subscribeOn(Schedulers.io())
    }

    override fun confirmToken(token: TransactionAuthenticationNumber): Single<VerificationPhoneResponse> {
        return verificationPhoneNetworkDataSource.postConfirm(token)
            .subscribeOn(Schedulers.io())
    }
}